homeassistant/components/flo/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""The flo integration."""
+import asyncio
+import logging
+
+from aioflo import async_get_api
+from aioflo.errors import RequestError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .device import FloDeviceDataUpdateCoordinator
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the flo component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up flo from a config entry."""
+    hass.data[DOMAIN][entry.entry_id] = {}
+    session = async_get_clientsession(hass)
+    try:
+        hass.data[DOMAIN][entry.entry_id]["client"] = client = await async_get_api(
+            entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD], session=session
+        )
+    except RequestError:
```

Comment:
Are you able to differentiate between incorrect auth and unable to reach server ?

Reply:
not at the moment. I'll do some testing w/ that and pr a patch to the lib if I can do something